### ISSUE TITLE
Class name stripping after dollar sign $

### ISSUE THREAD

**Issue by q3769** [tinylog-org/tinylog#380]
https://github.com/tinylog-org/tinylog/blob/eab5137437ecfdc7fb650105b970465b642e0f7c/tinylog-api/src/main/java/org/tinylog/runtime/RuntimeProvider.java#L229

Not common but a nested/inner class name is allowed to start with a lower case.

Also, I didn't get to read into the code in detail but is the class name stripping just for human readability or it has code logic significance? Seems unnecessary if just for human read. e.g. one might want to know a log message is issued from an anonymous inner class (whose name may be "XyzClass$1"); stripping off the "$1" may not be desired, or even misleading.

**Comment by pmwmedia** [comment]
> Not common but a nested/inner class name is allowed to start with a lower case.

You are right. However, this is the best heuristic we could find so far. I have never seen any class not starting with an upper case letter in a real world project so far. In tinylog 3, we only strip a nested/inner class name if it starts with a dollar sign or number.

> Also, I didn't get to read into the code in detail but is the class name stripping just for human readability or it has code logic significance?

It is mainly for human readability. However, it has also some indirect impact on setting specific severity levels for classes.

Actually, stripping class names is a very demanded feature and should be the default behavior. Nevertheless I could imagine to move the stripping logic to the class placeholder and make it configurable.
